brave-instrumentation/aws-java-sdk-core/src/test/java/brave/instrumentation/aws/MockS3Server.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2016-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.instrumentation.aws;
+
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public class MockS3Server implements TestRule {
```

Comment:
I see this is following the pattern of `MockDynamoDbServer`, but it's the exact same implementation, and both provide no value on top of prepending `http://localhost:` while hiding potentially useful APIs like `takeRequest`.

I think you can go ahead and use `MockWebServer` directly instead of making this class and add the `localhost:` in the test code. Bonus points to do similar for dynamo db.